### PR TITLE
Add or Remove Curator system tests

### PR DESCRIPTION
**Pull request by mohsenno1**

### CODE REVIEW

**Review by stewartie4 — APPROVED** [review]

File: tests/cli_tests/zboxcli_add_remove_curator_test.go [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+		require.Nil(t, err, "could not get allocation ID", err, strings.Join(output, "\n"))
+		defer createAllocationTestTeardown(t, allocationID)
+
+		params := createParams(map[string]interface{}{"allocation": allocationID})
+		output, err = addCurator(t, params)
+		require.NotNil(t, err, "unexpected success on adding curator", strings.Join(output, "\n"))
+		require.Equal(t, "Error: curator flag is missing", output[0], strings.Join(output, "\n"))
+	})
+
+	t.Run("Add Curator _ Curator must be able to transfer the allocation ownership", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_add_remove_curator_test.go:183]: very minor: but I'd move the add tests before remove

File: tests/cli_tests/zboxcli_add_remove_curator_test.go [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+		wallet, err := getWallet(t, configPath)
+		require.Nil(t, err, "Error occurred when retrieving wallet")
+
+		params := createParams(map[string]interface{}{"allocation": "INVALID ALLOCATION ID", "curator": wallet.ClientID})
+		output, err = addCurator(t, params)
+		require.NotNil(t, err, "unexpected success on adding curator", strings.Join(output, "\n"))
+		require.Contains(t, output[0], "adding curator:[txn] too less sharders to confirm it", strings.Join(output, "\n"))
+	})
+
+	t.Run("Add Curator _ must fail when the user is not the owner ", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_add_remove_curator_test.go:132]: This seems wrong to me
There is no such requirement that the curator must be the owner in the official docs.
Also if the only valid curator is the owner, why is there a "list" of curators (plural) 
The owner being the only valid curator defeats the purpose of the feature

File: tests/cli_tests/zboxcli_add_remove_curator_test.go [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+		require.Equal(t, curatorWallet.ClientID, allocation.Curators[0], "Curator must've added to the allocation curators list")
+
+		output, err = transferAllocationOwnershipWithWallet(t, allocationID, curatorWalletName, targetWallet.ClientID, targetWallet.ClientPublicKey)
+		require.Nil(t, err, "error in transferring allocation as curator", strings.Join(output, "\n"))
+		require.Equal(t, 1, len(output), "unexpected output length:", strings.Join(output, "\n"))
+		expectedOutput := fmt.Sprintf("transferred ownership of allocation %s to %s", allocationID, targetWallet.ClientID)
+		require.Equal(t, expectedOutput, output[0], "unexpected output:", strings.Join(output, "\n"))
+	})
+
+	t.Run("Add Curator _ Owner added as curator must be able to transfer the ownership", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_add_remove_curator_test.go:231]: This test is the same as above, worth keeping if we change the test above to not be the owner

File: tests/cli_tests/zboxcli_add_remove_curator_test.go [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+		require.Equal(t, "Error: curator flag is missing", output[0], strings.Join(output, "\n"))
+	})
+
+	t.Run("Add Curator _ Curator must be able to transfer the allocation ownership", func(t *testing.T) {
+		t.Parallel()
+
+		curatorWalletName := escapedTestName(t) + "_CURATOR"
+		targetWalletName := escapedTestName(t) + "_TARGET"
+
+		output, err := registerWallet(t, configPath)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_add_remove_curator_test.go:189]: We don't use this for anything in this test

File: tests/cli_tests/zboxcli_add_remove_curator_test.go (hunk 2 above)
Comment (stewartie4) [inline tests/cli_tests/zboxcli_add_remove_curator_test.go:132]: "Add Curator _ Curator must be able to transfer the allocation ownership" contradicts this test as a non-owner is added as a curator there

**Review by stewartie4 — CHANGES_REQUESTED** [review]
some queries

File: tests/cli_tests/zboxcli_add_remove_curator_test.go (hunk 2 above)
Comment (mohsenno1) [reply tests/cli_tests/zboxcli_add_remove_curator_test.go:132]: No, this seems to be like bad naming. Here by "user" I mean the operator which is adding the curator.
So if you are not the owner of an allocation then you cannot add someone else as the curator to that allocation.

Comment (mohsenno1) [reply tests/cli_tests/zboxcli_add_remove_curator_test.go:132]: I changed the test name to:

> Add Curator _ attempt to add curator by anyone except allocation owner must fail

File: tests/cli_tests/zboxcli_add_remove_curator_test.go (hunk 3 above)
Comment (mohsenno1) [reply tests/cli_tests/zboxcli_add_remove_curator_test.go:231]: The test are different. As I mentioned on the other comment, that is about testing if any one except allocation owner can add curator.
this is testing if after owner adds itself as curator is able to transfer the allocation.

File: tests/cli_tests/zboxcli_add_remove_curator_test.go (hunk 4 above)
Comment (mohsenno1) [reply tests/cli_tests/zboxcli_add_remove_curator_test.go:189]: you mean the wallet?
In this test we register a wallet and we create allocation by that. so this wallet is the owner. Then as the owner we add curatorWallet as curator. Then we test if curatorWallet is able to transfer allocation to a third party which is the targetWallet here.

**Pull request merged**